### PR TITLE
AK SQL Managed Instances

### PR DESCRIPTION
**Pull request by iriahk89**
Initial Release of msi thats include

- sql mi (default)
- sql managed db (default and PointInTimeRestore)
- Azure Ad Administrator
- FailoverGroup
- DB Backup Short Term Retention Policies

and also included standard mssql_server failovergroup

#106

### CODE REVIEW

File: examples/mssql_server/failover_groups/configuration.tfvars [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    databases = {
+      db1-rg1 = {
+        database_key = "db1-rg1"
+      }
+      db2-rg1 = {
+        database_key = "db2-rg1"
+      }
+    }
+    # primary_server_name = "test-sql-failover-source-server"
```
Comment (LaurentLesle) [inline examples/mssql_server/failover_groups/configuration.tfvars:68]: Cleanup those comments

File: modules/databases/mssql_managed_instance/administrator/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  required_version = ">= 0.13"
+}
+
+locals {
+  object_id    = coalesce(var.group_id, try(data.azuread_user.upn[0].id, null))
+  display_name = var.group_id != null ? var.group_name : var.user_principal_name
+
+  # arm_filename = "${path.module}/arm_mi_administrator.json"
+
+  # # this is the format required by ARM templates
```
Comment (LaurentLesle) [inline modules/databases/mssql_managed_instance/administrator/main.tf:16]: Cleanup please

File: modules/databases/mssql_managed_instance/administrator/mi_administrator.tf [hunk 3]
```diff
@@ -0,0 +1,61 @@
+
+data "azuread_user" "upn" {
+  count = try(var.user_principal_name, null) == null ? 0 : 1
+
+  user_principal_name = var.user_principal_name
+}
+
+# resource "azurerm_template_deployment" "miadministrator" {
```
Comment (LaurentLesle) [inline modules/databases/mssql_managed_instance/administrator/mi_administrator.tf:8]: Cleanup

File: modules/databases/mssql_managed_instance/administrator/arm_mi_administrator.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (LaurentLesle) [inline modules/databases/mssql_managed_instance/administrator/arm_mi_administrator.json:1]: Still using the ARM template to set admins? if not cleanup

**Review by LaurentLesle — CHANGES_REQUESTED** [review]
Very good first version. Thanks @iriahk89 
Can you address the requested changes?

**Review by arnaudlh — APPROVED** [review]

**Pull request merged**